<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>
	<arg name="imu_body" 	default="pelvis"/>
	<arg name="prefix" 	default="/ik_lowerbody_node"/>
	<arg name="origin" 	default="[0.0,0.0,0.0]"/>
	<arg name="tf_prefix" default=""/>
	<arg name="is_base_body" default="false"/>
	
	<arg name="heading" 	default="[0.0,0.0,-1.0]"/>
	<arg name="run_as_service" default="true"/>

	<group ns="$(arg imu_body)">
		<node name="pose_average" type="external_average_quaternion_node" pkg="osrt_ros" output="screen">
			<param name="origin"        type="yaml" value="$(arg origin)" />
			<param name="imu_raw" value="imu/$(arg imu_body)" />

			<!-- these 2 parameters are for showing the imu in the urdf correctly -->
			<param name="imu_cal_frame" value="$(arg tf_prefix)/$(arg imu_body)_imu" />
			<param name="imu_ref_frame" value="$(arg tf_prefix)/$(arg imu_body)" />
			<remap to="$(arg prefix)/$(arg imu_body)/imu_cal"					from="/imu_cal"/>
			<remap to="$(arg prefix)/$(arg imu_body)/avg_pose"					from="/avg_pose"/>
			<remap to="/ik/heading_angle"					from="/heading_angle"/>
			<param name="heading_ref_frame" value="subject_adds_heading2" />
		</node>
		<!--node pkg="osrt_ros" type="wth_node" name="wtfh" output="screen" if="$(arg is_base_body)" -->
		<node pkg="osrt_ros" type="wth_node" name="wtfh" output="screen">
		<!--node pkg="osrt_ros" type="wth_node" name="wtfh" if="$(arg is_base_body)" -->
			<param name="imu_base_measured_frame" value="imu/$(arg imu_body)_0"/>
			<param name="added_heading_frame" value="subject_adds_heading2"/>
			<rosparam param="origin">[0,0,0]</rosparam>	
			<param name="heading"    type="yaml" value="$(arg heading)" />
			<rosparam param="opensim_heading">[1,0,0]</rosparam>	
			<remap to="/ik/$(arg imu_body)/heading_angle"					from="/heading_angle"/>
			<param name="angle_offset" value="0"/>
			<param name="flip_sign" value="false"/>
			<param name="bypass_everything" value="false"/>
			<param name="angle_manual" value="0"/>
			
			<param name="run_as_service" value="$(arg run_as_service)"/>

			<param name="is_base_body" value="$(arg is_base_body)"/>
		</node>
	</group>
</launch>


